rpc/coreclient_test.go

```diff
@@ -0,0 +1,73 @@
+package rpc
+
+import (
+	"context"
+	"testing"
+
+	"github.com/celestiaorg/celestia-core/node"
+	"github.com/celestiaorg/celestia-node/testutils"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewClient(t *testing.T) {
+	_, backgroundNode := newClient(t)
+	//nolint:errcheck
+	backgroundNode.Stop()
+}
+
+func TestClient_GetStatus(t *testing.T) {
+	client, backgroundNode := newClient(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	//nolint:errcheck
+	t.Cleanup(func() {
+		backgroundNode.Stop()
+		cancel()
+	})
+
+	status, err := client.GetStatus(ctx)
+	require.Nil(t, err)
+	t.Log(status.NodeInfo)
+}
+
+func TestClient_StartBlockSubscription_And_GetBlock(t *testing.T) {
+	client, backgroundNode := newClient(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	//nolint:errcheck
+	t.Cleanup(func() {
+		backgroundNode.Stop()
+		cancel()
+	})
+
+	// make 3 blocks
+	err := client.Start()
+	require.Nil(t, err)
+
+	eventChan, err := client.StartBlockSubscription(ctx)
+	require.Nil(t, err)
+
+	for i := 0; i < 3; i++ {
+		<-eventChan
+	}
```

Comment:
Why don't we call `GetBlock` here on each event (with a `nil` height and an assert that the returned height matches the expected heigh) instead of only height 2 below (and only after unsubscribing)?